### DISCUSSION TITLE
How to parameterize scenario runs

### DISCUSSION THREAD

**Discussion by jkiviluo** [Spine-project/Spine-Toolbox#1710] [Support]
A question from @zhaokov  (moved from https://github.com/Spine-project/Spine-Toolbox/issues/1702)

In this sample project, I need to add a scenario factor vector to adjust the price. Below is the code for price adjusting:

#-------------------------
price_df_ini = pd.read_excel(flnm_in, sheet_name='price')

k = np.array([0.2, 0.4, 0.3]); #--- 1, 0.6, 0.5

price_df = price_df_ini;
price_df['Price'] = price_df['Price'] * k;
#-------------------------

I will create 100 different vectors between 0 and 1 to adjust the price, run the calculation for each, and get the average total cost of the 100 scenarios. I can use a "For" loop to handle 100 different factor vectors. However, I am very interested to use the function Spinetoolbox provides. I see you mentioned scenario analysis in the documentation, but it seems a little more complex than I need.

Here is my quick thought. I can generate 100 files (say, csv files) to respectively contain the 100 factor vectors and name the files as 1.csv, 2.csv, ..., 100.csv. This way, maybe I can indicate the file name ID (from 1 to 100) in a scenario table. I want to know how Spinetoolbox sets up a scenario analysis procedure to handle the 100 calculations parallelly? Is there a simple way to do my work? I appreciate any help and suggestion from you. Thank you again!

**Comment by jkiviluo** [comment]
I would output those different scenarios into single csv file. Could look like this:
Scen000,Alt000,elecNode,price,0
Scen001,Alt001,elecNode,price,0.01
...

They could then be imported to Spine DB using the importer. You would need one mapping to import scenario/alternatives and another mapping to import the data for each alternative. 

Then you would have those 100 scenarios (with the difference in data included) available and Toolbox could parallelize the runs (settings --> engine let's you limit the parallelization).

**Reply by zhaokov** [reply]
Thank you very much for setting up the new question and your instructions.    I have created a scenario csv file.  Please see the attachment.
[scenario.csv](https://github.com/Spine-project/Spine-Toolbox/files/8681949/scenario.csv)

I tried to follow your instruction by reading the documentation about import/export (https://spine-toolbox.readthedocs.io/en/latest/data_import_export.html), however I feel the documentation is not very clear for me to solve the problem.  Please see the screen shot below.  I am also thinking how to connect scenario to 'run_model' and generate the result file for each scenario.  

![image](https://user-images.githubusercontent.com/104167801/168151408-739d633b-6690-470a-ab4b-1d0650f85719.png)

Is there any example with some concrete guidance I can follow?  I am also attaching the project, in case you would need it.  Thank you again for your help! 
[wz_try_3_fixed_2.zip](https://github.com/Spine-project/Spine-Toolbox/files/8681966/wz_try_3_fixed_2.zip)

**Comment by zhaokov** [comment]
Sorry, I pasted a wrong screen-shot.  Here is a right one to show where I am.
![image](https://user-images.githubusercontent.com/104167801/168153467-63344f85-e5af-436d-a78b-72fab848bf7c.png)
